src/ConfigurationManager.ts

```diff
@@ -357,7 +358,16 @@ export class ConfigurationManager implements Disposable {
   private findServiceUrl(serviceType: string): string | undefined {
     switch (serviceType) {
       case Constants.SERVICE_TYPE_EXTERNAL: {
-        return this.getExternalUrl() + Constants.SERVICE_CHECK_PATH;
+        const url = new URL(
+          this.getExternalUrl() + Constants.SERVICE_CHECK_PATH,
+        );
+        const username = this.get("external.username");
+        const apiKey = this.get("external.apiKey");
+        if (username && apiKey) {
+          url.searchParams.set("username", username);
```

Comment:
If I'm reading this right, `username` and `apiKey` are being appended to the URL's query string. However, since these are sensitive items, they should not be part of the URL, since URLs are always passed in the clear and often wind up in various analytical logs. [LT's API](https://languagetool.org/http-api/swagger-ui/#!/default/post_check) spec lists those as `formData` fields, so they need passed as form data, not in the URL directly.

Reply:
Does this mean that the logic needs to move to the `buildServiceParameters()` method further below in this file?

The API accepts the parameters as URL parameters, too. Technically, this change works. Of course, it's much cleaner if these parameters are passed as `formData` .

What's the best way to move forward?

Reply:
I am not the original author of this PR, but I would like to see the changes incorporated. Based on the original PR, I created a new one, but this time passing them as form data: #561

Reply:
Sorry for taking so long to respond, and thank you all for chiming in. @thomaskrause - will review your PR this weekend (hopefully today). Thanks for stepping in!